src/components/UncontrolledTable.js

```diff
@@ -129,6 +131,34 @@ export default class UncontrolledTable extends React.Component {
     return isEqual(keys, newKeys);
   }
 
+  getVisibleRows = () => {
+    const { page } = this.state;
+    const { ascending, column } = this.state.sort;
+    const { paginated, pageSize, rows } = this.props;
+
+    const start = page * pageSize;
+    const end = start + pageSize;
+    const sortedRows = this.sortedData(rows, column, ascending);
+    return paginated ? sortedRows.slice(start, end) : sortedRows;
+  }
+
+  visibleRowsChanged(prevProps, prevState) {
+    const { page } = this.state;
+    const { ascending, column } = this.state.sort;
+    const { paginated, pageSize, rows } = this.props;
+
+    const { page: prevPage } = prevState;
+    const { ascending: prevAscending, column: prevColumn } = prevState.sort;
+    const { paginated: prevPaginated, pageSize: prevPageSize, rows: prevRows } = prevProps;
+
+    return page !== prevPage ||
+      ascending !== prevAscending ||
+      column !== prevColumn ||
+      paginated !== prevPaginated ||
+      pageSize !== prevPageSize ||
+      !this.isEqualUsingKeys(rows, prevRows);
```

Comment:
No actionable thought for this comment, but I wonder if doing `rows !== prevRows` would be enough as I would assume that the object reference would change if someone created a new object with new rows. Although someone can easily just do something like this:
```
<Table rows={{ ...myItems }} />
```
Which will trigger the onChange all the time 😄